### PR TITLE
add dart implementation

### PR DESCRIPTION
**Pull request by bipinkrish**
Added dart code, open to comments

### CODE REVIEW

File: src/dart/LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
+
```
Comment (AmitMY) [inline src/dart/LICENSE:2]: are you open to use the main license for this repo?
https://github.com/sign-language-processing/pose/blob/master/LICENSE.txt

File: src/dart/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: pose
+description: Dart library for loading, viewing, augmenting, and handling .pose files
+version: 1.1.5
+repository: https://github.com/bipinkrish/pose
```
Comment (AmitMY) [inline src/dart/pubspec.yaml:4]: should we move this away? if you want to keep it under your repository, maybe we should create a git sub-module that just links to your repo?

**Review by AmitMY — CHANGES_REQUESTED** [review]
This looks good to me, with a couple of comments.

Additionally, I would love it if we could run the tests in CI:
Could you please add a `dart.yaml` workflow file [here](https://github.com/sign-language-processing/pose/tree/master/.github/workflows) that will run the dart tests? Something like:
```yaml
name: Dart

on:
  pull_request:
    branches: [main]
  push:
    branches: [main]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: dart-lang/setup-dart@v1

      - run: dart pub get
      - run: dart format --output=none --set-exit-if-changed .
      - run: dart analyze
      - run: dart test
      ```

File: src/dart/LICENSE (hunk 1 above)
Comment (bipinkrish) [reply src/dart/LICENSE:2]: I could replace/remove the license.

File: src/dart/pubspec.yaml (hunk 2 above)
Comment (bipinkrish) [reply src/dart/pubspec.yaml:4]: That is for [pub.dev](pub.dev) to link a repository.

**Comment by bipinkrish** [comment]
> Could you please add a `dart.yaml` workflow file [here](https://github.com/sign-language-processing/pose/tree/master/.github/workflows?rgh-link-date=2024-03-19T08%3A16%3A13Z) that will run the dart tests? Something like:

On it.

File: src/dart/pubspec.yaml (hunk 2 above)
Comment (bipinkrish) [reply src/dart/pubspec.yaml:4]: > maybe we should create a git sub-module that just links to your repo?

Your call, either is ok for me.

**Comment by bipinkrish** [comment]
Added workflow, your call on sub-module

**Comment by bipinkrish** [comment]
I am sorry I messed it up

**Review by AmitMY — APPROVED** [review]
Looks good! Tests are passing

**Pull request merged**